src/App.js

```diff
@@ -1,11 +1,64 @@
 import React from 'react';
+import css from './App.module.css'
+import {createBrowserHistory} from "history";
+import {Switch, Redirect, Route, Router} from "react-router-dom";
+import {Dashboard} from "./components/Dashboard/Dashboard";
+import {SignIn} from "./components/SignIn/SignIn";
+import {SignUp} from "./components/SignUp/SignUp";
+import {CreateBoard} from "./components/CreateBoard/CreateBoard";
+import {Preloader} from "./components/Preloader/Preloader";
 
-function App() {
-    return (
-        <div>
-            This homework requires a Hero!
-        </div>
-    );
+export const customHistory = createBrowserHistory();
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loggedIn: false,
+            stateFetch: false
+        }
+    }
+
+    componentDidMount() {
+        const token = localStorage.getItem("TOKEN");
+        if(token) {
+            this.onChangeFlag(true);
+        }
+    }
+
+    onChangeFlag = (loggedIn) => {
+        this.setState({loggedIn})
+    };
+
+    isFetching = (fetching) => {
+        this.setState({
+            stateFetch: fetching
+        });
+    };
+
+    isEmpty = (str) => {
```

Comment:
Can be moved from the component (even in another file)

Reply:
done